core/src/main/java/jenkins/tasks/SimpleBuildStep.java

```diff
@@ -93,28 +93,76 @@ default void perform(@NonNull Run<?, ?> run, @NonNull FilePath workspace, @NonNu
 
     /**
      * Run this step.
+     * <p>
+     * This method <strong>must</strong> be overridden when this step requires a workspace context. If such a context is
+     * <em>not</em> required, it does not need to be overridden; it will then forward to
+     * {@link #perform(Run, EnvVars, TaskListener)}.
+     *
      * @param run a build this is running as a part of
      * @param workspace a workspace to use for any file operations
      * @param env environment variables applicable to this step
      * @param launcher a way to start processes
      * @param listener a place to send output
+     * @throws AbstractMethodError if this step requires a workspace context and neither this method nor {@link #perform(Run, FilePath, Launcher, TaskListener)} is overridden
      * @throws InterruptedException if the step is interrupted
      * @throws IOException if something goes wrong; use {@link AbortException} for a polite error
      * @since TODO
```

Comment:
Side note: I thought these were supposed to be updated as part of the release process? This particular method was added in 2.241 I think.

Reply:
(This pointed to the `@since TODO` but apparently does not get updates by GH when more commits are added)

Reply:
Not sure, @daniel-beck?

Reply:
Contributors update them every now and then, it’s not automated